### PR TITLE
[T4][T01-T3] 

### PR DESCRIPTION
**Pull request by yogamurti**

### CODE REVIEW

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -20,172 +20,228 @@ By: `Team SE-EDU`      Since: `Jun 2016`      Licence: `MIT`
...
-
-* *`list`* : lists all contacts
-* **`add`**`n/John Doe p/98765432 e/johnd@example.com a/John street, block 123, #01-01` : adds a contact named `John Doe` to the Address Book.
-* **`delete`**`3` : deletes the 3rd contact shown in the current list
-* *`exit`* : exits the app
-
+* *`add projectX`* : adds a task with *projectX* as its title
+* *`list`* : lists all tasks
+* *`delete 2`* : deletes the 2nd task shown in the current list
+* *`quit`* : quits 2Do
```
Comment (jeffryhartanto) [inline docs/UserGuide.adoc]: Detailed instruction, good.

File: docs/UserGuide.adoc [hunk 2]
```diff
@@ -20,172 +20,228 @@ By: `Team SE-EDU`      Since: `Jun 2016`      Licence: `MIT`
...
-=== Viewing help : `help`
+Dates
+****
+* Format 1: ddmmyy +
+`Example: 050417`
+* Format 2: day month +
+`Example: 5 jan`
+* Format 3: month day +
+` Example: dec 31`
+****
```
Comment (jeffryhartanto) [inline docs/UserGuide.adoc]: No year for second and third format?

File: docs/UserGuide.adoc [hunk 3]
```diff
@@ -20,172 +20,228 @@ By: `Team SE-EDU`      Since: `Jun 2016`      Licence: `MIT`
...
 
-Adds a person to the address book +
-Format: `add n/NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS [t/TAG]...`
+Reccurence
+****
+* Format 1: days / hours +
+`Example: 4 days`
+* Format 2: keyword +
+`Example: daily / weekly / yearly / annually`
```
Comment (jeffryhartanto) [inline docs/UserGuide.adoc]: May be it can be added in the `add command` section, as it's not a common parameter

File: docs/UserGuide.adoc [hunk 4]
```diff
@@ -20,172 +20,228 @@ By: `Team SE-EDU`      Since: `Jun 2016`      Licence: `MIT`
...
+* Floating tasks can be added with date, time field blank
+* For urgent tasks, u/true can be added. Or else, field can be left blank
+****
 
-=== Editing a person : `edit`
+Format 2: `add [-h] {title} on {date} at {time}` +
+Example: `add project on friday at 2359`
+****
+* Supports keywords _on_, _at_ and _days_
+****
```
Comment (jeffryhartanto) [inline docs/UserGuide.adoc]: Better create a sub section for adding each kind of task

File: docs/UserGuide.adoc [hunk 5]
```diff
@@ -20,172 +20,228 @@ By: `Team SE-EDU`      Since: `Jun 2016`      Licence: `MIT`
...
-* At least one of the optional fields must be provided.
-* Existing values will be updated to the input values.
-* When editing tags, the existing tags of the person will be removed i.e adding of tags is not cumulative.
-* You can remove all the person's tags by typing `t/` without specifying any tags after it.
+* Deletes task at that index (positive number) in the most recent call of list
 ****
 
-Examples:
+Format 2: `delete [-h]  {title} on {date} at {time}` +
+Example: `delete CS2103 Project V0.1`
```
Comment (jeffryhartanto) [inline docs/UserGuide.adoc]: Where is the date and time in the example?

File: docs/UserGuide.adoc [hunk 6]
```diff
@@ -20,172 +20,228 @@ By: `Team SE-EDU`      Since: `Jun 2016`      Licence: `MIT`
...
+****
+* Undos the last successful action executed
+****
 
-Deletes the specified person from the address book. Irreversible. +
-Format: `delete INDEX`
+=== Editing a task : `edit / -e`
 
+Format: `edit [-h] {index} m/[new main] d/[new date] n/[new note] tag/[new tags] a/[alarm] r/[recurrence] u/[urgent]` +
+Example: `edit 2 m/CS2103`
```
Comment (jeffryhartanto) [inline docs/UserGuide.adoc]: Better to also provide an example of the command with all params (most complex).

File: docs/UserGuide.adoc [hunk 7]
```diff
@@ -20,172 +20,228 @@ By: `Team SE-EDU`      Since: `Jun 2016`      Licence: `MIT`
...
+* Default reminder alarm timing before the deadline can be edited
+** Range value for reminders: 1 minute <= reminder <= 7 days
+** Default setting is 7 days
+* Default prompt message for reminders can be edited
+** Default message is _{main} needs to be completed soon!_
+* Default automatic mark for completion setting can be edited 
+** Default setting will prompt user requesting if the activity was completed
+** This can be changed to automatically marking the activity as complete instead of prompting
+** Alternatively, the prompt can be removed and the activity would be marked as missed. You can then manually mark the completed tasks
+****
```
Comment (jeffryhartanto) [inline docs/UserGuide.adoc]: Too many bullet points, try to find a better way to format all these info.

File: docs/UserGuide.adoc [hunk 8]
```diff
@@ -20,172 +20,228 @@ By: `Team SE-EDU`      Since: `Jun 2016`      Licence: `MIT`
...
+|find / -f|find [-h] {keyword}|
+|undo / -u|undo|
+|edit / -e|edit [-h] {index} m/[new main] d/[new date] n/[new note] tag/[new tags] a/[alarm] r/[recurrence] u/[urgent]|
+|view / -v|view [-h] {index}|
+|mark / -m|mark {index}|
+|unmark / -un|unmark {index}|
+|option/ -o|option a/[alarm] p/ [prompt] m/ [mark completion automatically]|
+|save / -s|save [directory]|
+|clear / -c|clear|
+|quit / -q|quit|
```
Comment (jeffryhartanto) [inline docs/UserGuide.adoc]: May be you can also include the examples.

File: README.adoc [hunk 9]
```diff
@@ -8,20 +8,19 @@ https://coveralls.io/github/se-edu/addressbook-level4?branch=master[image:https:
...
-* It is *written in OOP fashion*. It provides a *reasonably well-written* code example that is *significantly bigger* (around 6 KLoC)than what students usually write in beginner-level SE modules.
-* What's different from https://github.com/se-edu/addressbook-level3[level 3]:
-** A more sophisticated GUI that includes a list  panel and an in-built Browser.
-** More test cases, including automated GUI testing.
-** Support for _Build Automation_ using Gradle and for _Continuous Integration_ using Travis CI.
+
+* 2Do is a To Do List desktop application. It has a GUI but most of the user interactions happen using a CLI (Command Line Interface).
+* This is designed to allow users to efficiently manage their tasks.
+* It is written in OOP fashion. It provides a reasonably well-written code example that is significantly bigger (around 6 KLoC)than what students usually write in beginner-level SE modules.
+* This project is initiated by the sample project created by SE-EDU.
```
Comment (jeffryhartanto) [inline README.adoc]: Shouldn't it be in acknowledgements section?

File: docs/AboutUs.adoc [hunk 10]
```diff
@@ -6,55 +6,36 @@ We are a team based in the http://www.comp.nus.edu.sg[School of Computing, Natio
...
-'''
-
-=== http://github.com/lejolly[Joshua Lee]
-image::JoshuaLee.jpg[width="150", align="left"]
+=== https://github.com/radneran[V Narendar Nag]
+image::NarenV.jpg[width="150", align="left"]
 
 Role: Developer +
-Responsibilities: UI
+Responsibilities: Team Leader, Scheduling and Tracking, Model
```
Comment (jeffryhartanto) [inline docs/AboutUs.adoc:13]: Better put the component in the new line.

**Comment by jeffryhartanto** [comment]
@yogamurti @radneran @maltiso @teoshuqi 
Some comments added. Please close the PR after reading comments.

File: README.adoc (hunk 9 above)
Comment (yogamurti) [reply README.adoc]: Yup. Resolved.

File: docs/UserGuide.adoc (hunk 3 above)
Comment (yogamurti) [reply docs/UserGuide.adoc]: _edit command_ uses it too

File: docs/UserGuide.adoc (hunk 5 above)
Comment (yogamurti) [reply docs/UserGuide.adoc]: Removed this format

File: docs/UserGuide.adoc (hunk 2 above)
Comment (yogamurti) [reply docs/UserGuide.adoc]: Resolved

File: docs/UserGuide.adoc (hunk 7 above)
Comment (yogamurti) [reply docs/UserGuide.adoc]: Deleted some bullet points, shortened some sentences

**Pull request closed without merging**